src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/external_repacking_adjuster.cpp

```diff
@@ -12,35 +12,59 @@
-        const auto& out = shape_infer_consumers.empty() ? param->get_output_port(0)
-                                                        : shape_infer_consumers.back()->get_output_port(0);
-        const auto consumers = out.get_connected_ports();
-
-        for (const auto& consumer : consumers) {
-            auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(consumer.get_expr()->get_node());
-            if (brgemm && brgemm_utils::with_repacking(brgemm->get_type()) && consumer.get_index() == 1) {
-                const auto src_prc = brgemm->get_input_element_type(0);
-                const auto wei_prc = brgemm->get_input_element_type(1);
-                const auto isa = brgemm_utils::get_primitive_isa(src_prc, brgemm_utils::with_amx(brgemm->get_type()));
-                const auto inner_n_block = brgemm_utils::repacking::compute_inner_n_block(wei_prc);
-                const auto is_transposed_b =
-                    BrgemmCopyB::is_transposed(m_configurator->get_io_descs()[i]->get_layout());
-                auto config = BrgemmCopyBKernelConfig(src_prc, wei_prc, isa, false, is_transposed_b, inner_n_block);
-                m_executors[i] = std::make_shared<BrgemmCopyBKernelExecutor>(configurator->get_cache(), config);
+    for (const auto& [idx, _] : repacked_inputs_config) {
+        OPENVINO_ASSERT(idx < params.size(), "Incorrect index of repacked input");
+
+        m_executors[idx] = create_executor(params[idx], configurator->get_cache());
+        OPENVINO_ASSERT(m_executors[idx], "Repacked input executor must be inited!");
+    }
+    OPENVINO_ASSERT(repacked_inputs_config.size() == m_executors.size(), "Incorrect count of repacked inputs");
+}
+
+BrgemmExternalRepackingAdjuster::RepackExecutorPtr BrgemmExternalRepackingAdjuster::create_executor(
+    const ov::snippets::lowered::ExpressionPtr& param,
+    const ov::intel_cpu::MultiCacheWeakPtr& cache) {
+    const auto& shape_infer_consumers = ov::snippets::utils::get_first_child_shape_infer_expr_seq(param);
+    const auto& out =
+        shape_infer_consumers.empty() ? param->get_output_port(0) : shape_infer_consumers.back()->get_output_port(0);
+    const auto consumers = out.get_connected_ports();
+    for (const auto& consumer : consumers) {
+        auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(consumer.get_expr()->get_node());
+        if (brgemm && brgemm_utils::with_repacking(brgemm->get_type()) && consumer.get_index() == 1) {
+            const auto src_prc = brgemm->get_input_element_type(0);
+            const auto wei_prc = brgemm->get_input_element_type(1);
+            const auto isa = brgemm_utils::get_primitive_isa(src_prc, brgemm_utils::with_amx(brgemm->get_type()));
+            const auto inner_n_block = brgemm_utils::repacking::compute_inner_n_block(wei_prc);
+
+            ov::snippets::VectorDims layout;
+            if (!shape_infer_consumers.empty()) {
```

Comment:
It seems like we use the WA described in CVS-160048. If it so, could you please add a comment with the corresponding comment?